src/handlers/initiateDataRequest.ts

```diff
@@ -15,13 +20,29 @@ export const handler = async (
   }
 
   const validatedZendeskRequest = validateZendeskRequest(event.body)
+
   if (!validatedZendeskRequest.isValid) {
     return await handleInvalidRequest(event.body, validatedZendeskRequest)
   }
 
-  const messageId = await sendInitiateDataTransferMessage(
+  const requestParams =
     validatedZendeskRequest.dataRequestParams as DataRequestParams
-  )
+
+  try {
+    if (!(await matchZendeskTicket(requestParams))) {
```

Comment:
As we'll only use this function here I'd switch the boolean response around so you don't need to reverse it in the if block.  Renaming it something like zendeskTicketDiffersToRequest lets you make this a bit simpler.